### PR TITLE
Do not XHR intercept first party (1P) CDN resources

### LINKED ISSUE #16601: AMP4EMAIL: Don't proxy XHRs to CDN resources
E.g. `ww.js`

`b/111214711`

### PR DESCRIPTION
**Pull request by alabiaga**
fixes #16601 

- fetch URLs to a AMP HTML CDN will not be xhr intercepted to the viewer.

### CODE REVIEW

File: src/service/xhr-impl.js [hunk 1]
```diff
@@ -185,7 +187,8 @@ export class Xhr {
     }
     const viewer = Services.viewerForDoc(this.ampdocSingle_);
     const whenFirstVisible = viewer.whenFirstVisible();
-    if (!viewer.hasCapability('xhrInterceptor')) {
+    const firstPartyResource = urls.cdnProxyRegex.test(new URL(input).origin);
```
Comment (choumx) [inline src/service/xhr-impl.js]: `URL` is not supported everywhere. Use our helper functions in `url.js`.

Comment (alabiaga) [reply src/service/xhr-impl.js]: done. thanks.

File: src/service/xhr-impl.js [hunk 2]
```diff
@@ -185,7 +187,9 @@ export class Xhr {
     }
     const viewer = Services.viewerForDoc(this.ampdocSingle_);
     const whenFirstVisible = viewer.whenFirstVisible();
-    if (!viewer.hasCapability('xhrInterceptor')) {
+    const firstPartyResource =
+        urls.cdnProxyRegex.test(parseUrlDeprecated(input).origin);
```
Comment (jridgewell) [inline src/service/xhr-impl.js]: Use `isProxyOrigin` from `src/url.js`.

File: src/service/xhr-impl.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ import {fromIterator} from '../utils/array';
 import {
   getCorsUrl,
   getSourceOrigin,
-  getWinOrigin,
```
Comment (jridgewell) [inline src/service/xhr-impl.js]: Why was this deleted?

Comment (alabiaga) [reply src/service/xhr-impl.js]: argh. was not intended. reverted.

**Comment by alabiaga** [comment]
THANKS!
